app/cdap/components/GridTable/components/NestedMenu/index.tsx

```diff
@@ -0,0 +1,149 @@
+        const updatedAnchors = anchorElement.splice(1, 0, event.currentTarget);
+        setAnchorElement((prev) => anchorElement);
+        setMenuComponentOptions([menuItem?.options]);
+      } else {
+        submitMenuOption(menuItem.value, menuItem.supportedDataType);
+        setAnchorElement(null);
+        menuToggleHandler(title);
+      }
+    } else {
+      if (menuItem.hasOwnProperty('options') && menuItem?.options?.length > 0) {
+        let referenceIndex = -1;
+        menuComponentOptions.forEach((eachMenuOptions, menuOptionsIndex) => {
+          eachMenuOptions.forEach((eachOption, optionsIndex) => {
+            if (eachOption.value === menuItem.value) {
+              referenceIndex = menuOptionsIndex;
+            }
+          });
+        });
+        if (referenceIndex >= 0 && !anchorElement.includes(event.currentTarget)) {
+          const insertPosition = referenceIndex + 2;
+          anchorElement.splice(insertPosition, 0, event.currentTarget);
+          const updatedAnchors = anchorElement.slice(0, insertPosition + 1);
+          menuComponentOptions.splice(updatedAnchors.length - 2, 0, menuItem?.options);
+          setMenuComponentOptions((prev) =>
+            prev.length ? menuComponentOptions : [menuItem?.options]
+          );
+          setAnchorElement(updatedAnchors);
+        } else if (anchorElement.includes(event.currentTarget)) {
+          const currentTargetIndex = _.findIndex(
+            anchorElement,
+            (anchor) => anchor === event.currentTarget
+          );
+          menuComponentOptions.splice(currentTargetIndex, 0, menuItem?.options);
+          setAnchorElement((prev) => prev.slice(0, currentTargetIndex + 1));
+          setMenuComponentOptions((prev) => menuComponentOptions.slice(0, currentTargetIndex + 1));
+        } else {
+          setAnchorElement((prev) => [...prev, event.currentTarget]);
+          setMenuComponentOptions((prev) =>
+            prev.length ? [...prev, menuItem?.options] : [menuItem?.options]
+          );
+        }
+      } else {
+        submitMenuOption(menuItem.value, menuItem.supportedDataType);
+        setAnchorElement(null);
+        menuToggleHandler(title);
+      }
+    }
+  };
+  return (
+    <>
+      <Menu
+        id="parent-menu"
+        data-testid="nested-menu-parent-root"
+        keepMounted
+        anchorEl={anchorElement?.length ? anchorElement[0] : null}
+        open={anchorElement?.length ? true : false}
+        getContentAnchorEl={null}
+        anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+        transformOrigin={{ vertical: 'top', horizontal: 'left' }}
+        onClose={() => menuToggleHandler()}
+        onClick={(clickEvent) => {
+          clickEvent.preventDefault();
+          clickEvent.stopPropagation();
+        }}
+        className={classes.root}
+        classes={{ paper: classes.popoverPaper }}
+      >
+        {Array.isArray(menuOptions) &&
+          menuOptions.length &&
```

Comment:
Why check for `Array.isArray(menuOptions)`? It's already asserted via the type system. Also why `menuOptions.length`? `map` will behave predictably on a 0-length array.

Reply:
Done